From cf94772e2564109579c43db88a472eeb6c31350e Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <contact@chrisaw.io>
Date: Wed, 10 Jun 2026 12:36:10 +0100
Subject: [PATCH 3/3] sepolicy: label dmabuf filesystem to fix camera video
 recording

on some vendor kernels, the dma-buf heap is exposed as a mountable
filesystem type ("dmabuf"). untrusted apps (including the camera)
need to statfs this filesystem to create PersistentSurface-backed video
encoder input surfaces, but SELinux blocks getattr on the unlabeled
filesystem:

  avc: denied { getattr } for name="/" dev="dmabuf"
    scontext=u:r:untrusted_app:s0 tcontext=u:object_r:unlabeled:s0
    tclass=filesystem permissive=0

label the dmabuf filesystem as rootfs (a pseudo-filesystem type that
apps already have getattr access to via the appdomain neverallow
exception for fs_type:filesystem:getattr).

this fixes the SELinux denial that prevents camera video recording on
devices with vendor kernels that expose /dev/dmabuf as a filesystem.

fixes: #80
---
 private/genfs_contexts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/private/genfs_contexts b/private/genfs_contexts
index bc48042..a9240ee 100644
--- a/private/genfs_contexts
+++ b/private/genfs_contexts
@@ -125,6 +125,7 @@ genfscon fusectl / u:object_r:fusectlfs:s0
 genfscon selinuxfs / u:object_r:selinuxfs:s0
 genfscon cgroup / u:object_r:cgroup:s0
 genfscon cgroup2 / u:object_r:cgroup_v2:s0
+genfscon dmabuf / u:object_r:rootfs:s0
 # sysfs labels can be set by userspace.
 genfscon sysfs / u:object_r:sysfs:s0
 genfscon sysfs /bus/thunderbolt/devices           u:object_r:sysfs_thunderbolt:s0
-- 
2.54.0

